app/forms/admin_reservation_form.rb

```diff
@@ -0,0 +1,72 @@
+class AdminReservationForm
+
+  include ActiveModel::Validations
+
+  REPEAT_OPTIONS = %w(
+    daily
+    weekdays_only
+    weekly
+    monthly
+  ).freeze
+
+  delegate :to_key, :to_model, to: :reservation
+  delegate :category, :reserve_start_date, :reserve_start_hour,
+           :reserve_start_min, :reserve_start_meridian, :duration_mins,
+           :reserve_end_date, :reserve_end_hour, :reserve_end_min,
+           :reserve_end_meridian, :admin_note, :expires?,
+           :expires_mins_before, to: :reservation
+  delegate :assign_times_from_params, to: :reservation
+
+  attr_accessor :repeats, :repeat_frequency, :repeat_end_date
+  attr_reader :reservation
+
+  # validates :max_end_date
+
+  def initialize(reservation)
+    @reservation = reservation
+  end
+
+  def assign_attributes(attrs)
+    @reservation.assign_attributes(attrs.except(:repeat_frequency, :repeat_end_date))
+    @repeat_frequency = attrs[:repeat_frequency]
+    @repeat_end_date = attrs[:repeat_end_date]
+  end
+
+  def save
+    if [@reservation, self].map(&:valid?).all?
+      @reservation.save
+      create_recurring_reservations
+    else
+      self.errors.merge(@registration.errors)
+      false
+    end
+  end
+
+  def create_recurring_reservations
```

Comment:
So this is super WIP-y, feel free to totally restructure this entire create_recurring business if you do end up picking this up, there are probably much better ways of doing this than my half-formed thought here.